### ISSUE TITLE
getting input/output dtype as  float32  after converting  keras mnist model to integer quantized tflite model 

### ISSUE THREAD

**Issue by shalushajan95** [tensorflow/tensorflow#42092]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):ubuntu 16.04
- TensorFlow installed from (source or binary): 
- TensorFlow version (or github SHA if from source): 2.2

**Command used to run the converter or code if you’re using the Python API**
If possible, please share a link to Colab/Jupyter/any notebook.

```
# Copy and paste here the exact command
```
import tensorflow as tf 
import numpy as np

# Load MNIST dataset
mnist = tf.keras.datasets.mnist

(train_images, train_labels), (test_images, test_labels) = mnist.load_data()

# Normalize the input image so that each pixel value is between 0 to 1.
train_images = train_images.astype(np.float32) / 255.0
test_images = test_images.astype(np.float32) / 255.0

# Define the model architecture
model = tf.keras.Sequential([
  tf.keras.layers.InputLayer(input_shape=(28, 28)),
  tf.keras.layers.Reshape(target_shape=(28, 28, 1)),
  tf.keras.layers.Conv2D(filters=12, kernel_size=(3, 3), activation='relu'),
  tf.keras.layers.MaxPooling2D(pool_size=(2, 2)),
  tf.keras.layers.Flatten(),
  tf.keras.layers.Dense(10)
])

# Train the digit classification model
model.compile(optimizer='adam',
              loss=tf.keras.losses.SparseCategoricalCrossentropy(
                  from_logits=True),
              metrics=['accuracy'])
model.fit(
  train_images,
  train_labels,
  epochs=5,
  validation_data=(test_images, test_labels)
)
model.save("mnist.h5")

model_ = tf.keras.models.load_model("mnist.h5")

def representative_data_gen():
  for input_value in tf.data.Dataset.from_tensor_slices(train_images).batch(1).take(100):
    # Model has only one input so each data point has one element.
    yield [input_value]

converter = tf.lite.TFLiteConverter.from_keras_model(model_)
converter.optimizations = [tf.lite.Optimize.DEFAULT]
converter.representative_dataset = representative_data_gen
# Ensure that if any ops can't be quantized, the converter throws an error
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS_INT8]
# Set the input and output tensors to uint8 (APIs added in r2.3)
converter.inference_input_type = tf.uint8
converter.inference_output_type = tf.uint8

tflite_model_quant = converter.convert()

interpreter = tf.lite.Interpreter(model_content=tflite_model_quant)
input_type = interpreter.get_input_details()[0]['dtype']
print('input: ', input_type)
output_type = interpreter.get_output_details()[0]['dtype']
print('output: ', output_type)
**The output from the converter invocation**

```
# Copy and paste the output here.

<class 'numpy.float32'>
<output:  <class 'numpy.float32'>

```

**Also, please include a link to the saved model or GraphDef**

```
# Put link here or attach to the issue.
```

**Failure details**
If the conversion is successful, but the generated model is wrong,
state what is wrong:
- Producing wrong results and/or decrease in accuracy
- Producing correct results, but the model is slower than expected (model generated from old converter)

**RNN conversion support**
If converting TF RNN to TFLite fused RNN ops, please prefix [RNN] in the title.

**Any other info / logs**

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

Hi 
Please help me to identify this issue, After successful conversion of mnist model to integer quantized tflite model i am getting the input dtype as float32 instead  of uint8.
I followed the same post training interger quantization steps mentioned in the tensorflows official website, but getting input/output dtype  as float32
Attaching the link: https://www.tensorflow.org/lite/performance/post_training_integer_quant

I also followed the command line tflite_converter command too,  still getting quantized float model only
command used : 
tflite_convert --output_file mnist_tflite_cmd.tflite --keras_model_file mnist.h5 --input_arrays "reshape_input" --input_shapes "1,28,28" --output_arrays "Identity" --output_format TFLITE --inference_type QUANTIZED_UINT8 --inference_input_type QUANTIZED_UINT8 

Thanks

**Comment by Saduf2019** [comment]
@shalushajan95 
Please provide indented code such that we can replicate the issue or if possible share a colab gist with the error reported.

**Comment by shalushajan95** [comment]
@Saduf2019 
Code is attached 
Than you
[minst_tflite_converter.py.zip](https://github.com/tensorflow/tensorflow/files/5039527/minst_tflite_converter.py.zip)

**Comment by Saduf2019** [comment]
@shalushajan95 
I ran the code shared and i am able to get unit8 [on tf 2.3]  as expected, please find the [gist here](https://colab.research.google.com/gist/Saduf2019/d0f3c92e2b94fee80139339e28075735/untitled335.ipynb).
In case you are still facing the issue on tf 2.3 please share a colab gist.
